login-module/src/index.js

```diff
@@ -8,52 +8,34 @@ import misc from './misc';
 window.GraphAcademyLogin = class GraphAcademyLogin {
 	constructor(options = {}) {
 		this.options = { ...constants.DEFAULT_OPTIONS, ...options };
-		const hasRequiredOptions = this.hasRequiredOptions(options);
+		const hasRequiredOptions = this.hasRequiredOptions(this.options);
 		if (!hasRequiredOptions) {
 			console.log(`required params missing - one of ${constants.REQUIRED_OPTIONS.join(', ')}`);
 			return;
 		}
-		this.webAuth = new WebAuth({
-			clientID: 'hoNo6B00ckfAoFVzPTqzgBIJHFHDnHYu',
-			domain: 'login.neo4j.com',
-			redirectUri: `${window.location.origin}/accounts/login`,
-			audience: 'neo4j://accountinfo/',
-			scope: 'read:account-info openid email profile user_metadata',
-			responseType: 'token id_token'
-		});
+		this.core = new GraphAcademyCore(this.options);
+		this.certificate = new Certificate(this.options.trainingClassName, this.options.stage)
+		this.enrollment = new Enrollment(this.options.trainingClassName, this.options.stage)
+		this.quiz = new Quiz(this.options.trainingClassName, this.options.stage)
 		// User data
 		this.quizesStatus = [];
 		this.enrollmentStatus = [];
 		this.currentModule = '';
 		this.quizModuleCount = null;
 		this.currentModuleQuizStatus = null;
-		this.certificate = new Certificate(this.options.trainingClassName, this.options.stage)
-		this.enrollment = new Enrollment(this.options.trainingClassName, this.options.stage)
-		this.quiz = new Quiz(this.options.trainingClassName, this.options.stage)
 	}
 
 	hasRequiredOptions(options) {
 		return constants.REQUIRED_OPTIONS.every(item => options[item]);
 	}
 
-	checkSession(cb) {
-		const { options, webAuth } = this;
-		const { trainingClassName, stage } = options;
+	async checkSession(cb) {
+		const { options, core } = this;
 		misc.handleHtmlOnState('checkingSession', options);
-		webAuth.checkSession({}, async (err, result) => {
-
-			if (err) {
-				this.isLoggedIn = false;
-				if (options.loginRedirectUrl) {
-					this.redirectToLogin();
-				}
-			}
-
+		try {
+			const result = await core.checkSession();
```

Comment:
What would happen if the user is not logged in? Is the plan for the on page js to explicitly call the `redirectToLogin()`?

Reply:
The "redirect to login" feature is part of the core library if `loginRedirectUrl` is defined. But you can also opt-out and do something else when the user is not logged in:

```js
// in this example loginRedirectUrl option is not defined
try {
  const result = await core.checkSession();
} catch (e) {
  console.log('User is not logged in!', e)
  // do something!
}
```